Plugins/Json/include/Acts/Plugins/Json/JsonHierarchicalObjectConverter.hpp

```diff
@@ -0,0 +1,81 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryID.hpp"
+#include "Acts/Geometry/HierarchicalGeometryContainer.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <map>
+
+namespace Acts {
+
+/// Convert Hierarchical Object Container to Json file and vice-versa
+template <typename object_t>
+struct JsonHierarchicalObjectConverter {
+ public:
+  /// The object key
+  std::string datakey = "Object";
+
+  /// The loglevel
+  Acts::Logging::Level logLevel = Acts::Logging::Level::INFO;
+
+  /// Convert json map to Hierarchical Object Container
+  ///
+  /// @param map The indexed Hierarchical Object in json format
+  /// @param fromJson Function that return Hierarchical Object from json
+  HierarchicalGeometryContainer<object_t> jsonToHierarchicalContainer(
+      const nlohmann::json& map,
+      std::function<object_t(const nlohmann::json&, const Acts::GeometryID&)>
+          fromJson) const;
+
+  /// Convert json map to Hierarchical Object Container
+  ///
+  /// @param hObject The Hierarchical Object Container
+  /// @param toJson Function that convert Hierarchical Object to json
+  nlohmann::json hierarchicalObjectToJson(
+      const HierarchicalGeometryContainer<object_t>& hObject,
+      std::function<nlohmann::json(const object_t&)> toJson) const;
+
+ private:
+  /// The detector tag
+  std::string m_detkey = "detector";
```

Comment:
These values are fixed and do not need to be member variables. I think it is perfectly fine to have them as-is in the code. If you want to have them as constants, you can make the `static constexpr`.

Reply:
OK lets go for that ! While we are on the topic of name do you have any preference for the one using tracking geometry ?